src/matches.ts

```diff
@@ -0,0 +1,56 @@
+import { purry } from './purry';
+
+function _matches(target: any, condition: any) {
+  if (typeof condition === 'function') return condition(target);
+
+  if (typeof condition === 'object') {
+    return Object.entries(condition).every(([key, value]) =>
+      matches(target[key], value)
+    );
+  }
+
+  return target === condition;
```

Comment:
should we compare with `Object.is` here?

Reply:
I don't think so. The target can be anything that is not an object or a function at that point.

Reply:
Object.is is not just for objects, it supports any JS thing and it might be a little more accurate for a general purpose function like this, but it's also es2015 so using === would support more users (though we probably want to move to es2015 anyway).

Here's the answer from chatgpt :P

> Object.is:
Object.is is a method introduced in ECMAScript 6 (ES6) that compares two values for equality, similar to the strict equality operator (===). It returns true if the values are the same, both in terms of type and value. However, it has some special cases:
It considers +0 and -0 as not equal, unlike === which treats them as equal.
It considers NaN values as equal, unlike ===, which considers them not equal.
> In general, you can use === for most equality comparisons. However, if you specifically need to handle the special cases mentioned above, Object.is provides a more precise comparison.

Reply:
Sorry, I can't take seriously any argument quoting chatGPT. I just.. can't.

Reply:
I was suggesting `Object.is` because it is mostly equivalent to triple equals, except that it handles edge cases around NaN / zero better. It has nothing to do with the things to compare being an object. It is also what react uses under the hood for comparisons when updating state.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is